### PR TITLE
Update dg calendar implementation

### PR DESCRIPTION
**Pull request by syamfarh**
*calendar implementation
*sort implementation

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -17,8 +17,11 @@ FAPro - Developer Guide
     * [[Proposed]Undo/Redo feature](#proposed-undoredo-feature)
     * [Clone feature](#clone-feature)
     * [Undo feature](#undo-feature)
-    * [[Proposed]Dara archiving](#proposed-data-archiving)
-    * [Find feature](#find-by-address-feature)
+    * [Find feature](#find-feature)
```
Comment (bhnuka) [inline docs/DeveloperGuide.md:21]: could add for my delete command? ;")

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -570,6 +567,55 @@ Step 7. A list of all contacts who have `2023-12-12` matching their appointment
...
+Step 5: Then, `SortCommand#execute` is called, which call `Model#updateSortComparator(Comparator)` to update the Comparator in the Model.
+
+Step 6: The list of contacts is updated in ascending order by name.
+
+![Sort1](images/Sort1.png)
+
+Now, James is at index 1, John is at index 2, Luke is at index 3, Peter is at index
+4 and Simon is at index 5.
+
+![Sort2](images/SortActivityDiagram.png)
```
Comment (bhnuka) [inline docs/DeveloperGuide.md:787]: By right you can only have one end for the diagram... I'd recommend changing it. Maybe the invalid errors should lead to an if else to check if valid and if not go to end?

**Review by bhnuka — COMMENTED** [review]
Mostly alright! Just need you to help add the hyperlink for delete (oops) and also update the activity diagram for sort. Thanks!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (syamfarh) [reply docs/DeveloperGuide.md:787]: I actually dunno how to do it lel, lemme try!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (syamfarh) [reply docs/DeveloperGuide.md:21]: done!

**Pull request merged**